### DISCUSSION TITLE
Handling validation messages when binding with model properties

### DISCUSSION THREAD

**Discussion by travisobregon** [livewire/livewire#1489] [4. General]
Is there a better way to handle displaying validation messages when binding with model properties? 

From the example [here,](https://laravel-livewire.com/docs/2.x/properties#binding-models) an error message would be "The post.title field is required.", but you most likely want "The title field is required.". Am I missing something?

I have this temporary workaround:
```
$this->validate(
    $this->rules(),
    [],
    collect($this->rules())->map(fn ($_, $key) => Str::of($key)->afterLast('.')->replace('_', ' '))->toArray()
);
```

**Comment by affektde** [comment]
We tried to use Laravel's localization features:

```
Specifying Custom Attribute Values
If you would like the :attribute portion of your validation message to be replaced with a custom attribute name, you may specify the custom name in the attributes array of your resources/lang/xx/validation.php language file:

'attributes' => [
    'email' => 'email address',
],
```

but it did not catch for example `post.title` like in your example. Still using the code like you do above.
Would be nice to use Laravel's features here.

**Comment by zepfietje** [comment]
Seems to have landed in [v2.1.0](https://github.com/livewire/livewire/releases/tag/v2.1.0)!

**Reply by travisobregon** [reply]
It looks like there is still an issue if the property has an underscore, such as `job_title`

**Reply by zepfietje** [reply]
@travisobregon if so, just open an issue as a bug report.

**Reply by zepfietje** [reply]
@travisobregon, I've opened a bug report to track this issue: https://github.com/livewire/livewire/issues/1524.
